controllers/userController.js

```diff
@@ -24,7 +25,47 @@ monitor = {
 user = {
   create: async () => {}, //placeholder
   read: async () => {}, //palceholder
-  update: async () => {}, //placeholder
+  update: async (req, res) => {
+    try {
+      const user = req.user;
+      // get the data the user want to update
+      const data = {};
+      Object.keys(req.body).forEach((cur) => {
+        if (cur === 'password' || cur === 'name' || cur === 'email') {
+          data[cur] = req.body[cur];
+        }
+      });
+
+      //check if the new email exist in the db
+      let result;
+      if (data.email) {
+        result = await User.findOne({ email: data.email });
+        if (result) throw 'Email already exist';
+      }
+
+      //hash the password
+      let hashedPassword;
+      if (data.password) {
+        hashedPassword = await bcrypt.hash(data.password, 12);
+        data.password = hashedPassword;
+      }
+      // update
+      await User.findByIdAndUpdate(user._id, data, { runValidators: true });
+
+      //if password changed => logout
+      if (data.password) return res.redirect('auth/logout');
+
+      res.status(200).json({
+        status: 'success',
+        message: 'profil updated successfullu',
```

Comment:
Only for questions of spelling, we should update this message for 'Profile updated Successfully.